services/121-service/src/notifications/auth.middlewareTwilio.ts

```diff
@@ -12,20 +12,34 @@ export class AuthMiddlewareTwilio implements NestMiddleware {
     res: Response,
     next: NextFunction,
   ): Promise<void> {
-    const twilioSignature = req.headers['x-twilio-signature'];
-
     if (process.env.MOCK_TWILIO) {
       return next();
     }
 
+    let twilioSignature: string | string[] | undefined =
+      req.headers['x-twilio-signature'];
+    if (Array.isArray(twilioSignature)) {
+      twilioSignature = twilioSignature[0];
+    }
+    if (!twilioSignature) {
+      throw new HttpException(
+        'Twilio signature not found',
+        HttpStatus.UNAUTHORIZED,
+      );
+    }
+
+    if (!process.env.TWILIO_AUTHTOKEN) {
+      throw new HttpException(
+        'Twilio auth token not found',
+        HttpStatus.INTERNAL_SERVER_ERROR,
+      );
+    }
+
     const validWhatsAppStatus = twilio.validateRequest(
       process.env.TWILIO_AUTHTOKEN,
       twilioSignature,
       EXTERNAL_API.whatsAppStatus,
       req.body,
-      {
```

Comment:
`twilio.validateRequest` only accepts 4 parameters, so it seems that these extra parameters were ignored. TS was not complaining before, because `twilio` was typed as `any`, therefore we had no insight into what these functions expect.

Maybe a previous version of the twilio library expected this extra info.